src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/RunConcurrentRepresentationTests.kt

```diff
@@ -469,19 +468,25 @@ class CoroutinesRunConcurrentRepresentationTest : BaseRunConcurrentRepresentatio
         private val channel2 = Channel<Int>(capacity = 1)
     }
 
-    override fun block() = Executors.newFixedThreadPool(2).asCoroutineDispatcher().use { dispatcher ->
-        runBlocking(dispatcher) {
-            val job1 = launch(dispatcher) {
-                channel1.send(sharedCounter++)
-                r1 = channel2.receive()
-            }
-            val job2 = launch(dispatcher) {
-                channel2.send(sharedCounter++)
-                r2 = channel1.receive()
+    override fun block() {
+        val threadNumber = AtomicInteger(0)
+        val executorService = Executors.newFixedThreadPool(2) { runnable ->
+            Thread(runnable, "Thread-${threadNumber.incrementAndGet()}")
+        }
```

Comment:
You could add to the test utils a new class:
```kotlin
class DeterministicThreadNamesFactory(private val prefix: String) : ThreadFactory {
    private val threadNumber = AtomicInteger(0)

    override fun newThread(runnable: Runnable): Thread {
        val threadName = "$prefix-${threadCount.getAndIncrement()}"
        return Thread(runnable, threadName)
    }
}
```
and pass it in both places:

```kotlin
Executors.newFixedThreadPool(2, DeterministicThreadNamesFactory("Thread"))
```

It might be used elsewhere later